### PR TITLE
nginx.conf: support PATCH method

### PR DESCRIPTION
**Pull request by MayeulC**
## Problem

Could not update the EXIF metadata of my pictures:
https://github.com/pulsejet/memories/issues/341#issuecomment-1385419805

Nginx was returning a 405 when seeing the `PATCH` verb

## Solution

Handle the `PATCH` verb the same way others are handled.

Not sure if there are security considerations. It could be restricted to the path that is called: `/apps/memories/api/image/set-exif/123456xx`.

## PR Status

- [X] Code finished and ready to be reviewed/tested
- [X] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by MayeulC** [comment]
!testme

File: conf/nginx.conf [hunk 1]
```diff
@@ -123,6 +123,9 @@ location ^~ __PATH__/ {
     if ($request_method ~ ^DELETE$) {
         rewrite ^ __PATH__/index.php$request_uri last;
     }
+    if ($request_method ~ ^PATCH$) {
```
Comment (ericgaspar) [inline conf/nginx.conf:126]:
```
  location ~ / {
    if ($request_method ~ ^(PUT|DELETE|PATCH)$) {
        rewrite ^ __PATH__/index.php$request_uri last;
    }
    try_files $uri / __PATH__/index.php$request_uri;
  }
```

Comment (MayeulC) [reply conf/nginx.conf:126]: Good call. Though I'm mostly wondering if there are any security considerations to have, and if other keywords need to be supported.
I will make another commit with the suggested change.

**Comment by MayeulC** [comment]
!testme

**Comment by MayeulC** [comment]
I re-launched the CI manually some time ago, and it passed: https://ci-apps-dev.yunohost.org/ci/job/5265 That run was on c4b43f0add3684c170fd4356229c0f73a6ad1ec2.

However it didn't update the image above. Could this please be merged? Thanks in advance.

**Review by ericgaspar — APPROVED** [review]

**Pull request merged**